src/sidebar/components/search-status-bar.js

```diff
@@ -1,17 +1,21 @@
 'use strict';
 
-const memoize = require('../util/memoize');
+const { createElement } = require('preact');
+const propTypes = require('prop-types');
 
+const memoize = require('../util/memoize');
+const { withServices } = require('../util/service-context');
 const uiConstants = require('../ui-constants');
+const useStore = require('../store/use-store');
 
-// @ngInject
-function SearchStatusBarController(store, rootThread) {
-  this.TAB_ANNOTATIONS = uiConstants.TAB_ANNOTATIONS;
-  this.TAB_NOTES = uiConstants.TAB_NOTES;
-  this.TAB_ORPHANS = uiConstants.TAB_ORPHANS;
-
+function SearchStatusBar({
+  selectedTab,
+  totalAnnotations,
+  totalNotes,
+  rootThread,
+}) {
   const thread = () => {
-    return rootThread.thread(store.getState());
+    return rootThread.thread(useStore(store => store.getState()));
   };
 
   const visibleCount = memoize(thread => {
```

Comment:
I've just realized that the recursive use of `visibleCount` made the memoization here basically pointless because any recursive calls will have a different argument. To use memoization here you'd need to do two things:

1. Use the `useMemo` or `useCallback` hooks rather than `memoize`, because the hooks remember state across renders.
2. Separate `visibleCount` into a memoized function which takes a root thread as input and a non-memoized helper which is used for the recursive calls.

It might be helpful here to remove the memoization and get this working directly, then re-add it in a separate PR.